### PR TITLE
Make error alerts more screen reader-friendly

### PR DESCRIPTION
**Pull request by sannidhishukla**
# Overview

This PR adds some updates and improvements to error alerts on the AI assistant to make them more screen reader-friendly and reduce clutter on the page

- Remove the fadeout animation for error alerts
- Update the `aria-label` for error alerts to be shorter
- Update alert logic so that duplicate alerts are not shown
- Update alert text so that one alert can display information for multiple errors

Thank you to Lizzy, Sarah, Melanie, Andrew, and Amanda for the feedback and discussion to help improve these alerts!

# Screenshots
## Before
Screenshot 2025-03-18 at 2 02 42 PM

## After
Screenshot 2025-03-18 at 2 03 14 PM

### CODE REVIEW

File: frontend/jest.setup.tsx [hunk 1]
```diff
@@ -9,16 +9,7 @@ jest.mock("framer-motion", () => ({
...
-    children: React.ReactNode;
-    onExitComplete?: () => void;
-  }) => {
-    if (onExitComplete != null) {
-      setTimeout(() => {
-        onExitComplete();
-      }, 10000);
-    }
-
+  AnimatePresence: ({ children }: { children: React.ReactNode }) => {
```
Comment (casewalker) [inline frontend/jest.setup.tsx:12]: sand - Should this exist anymore? It seems like it is doing nothing for now, but I would be ok with leaving it in if there are concrete plans to change it again in the future.

File: frontend/src/components/QuestionInput/ErrorAlert.test.tsx [hunk 2]
```diff
@@ -4,31 +4,32 @@ import { axe, toHaveNoViolations } from "jest-axe";
...
-    const alertText = within(errorAlert).getByText(alert.message);
+    const alertText = within(errorAlert).getByText(alertMessage);
     expect(alertText).toBeInTheDocument();
+    expect(alertText).toHaveAttribute("role", "alert");
 
     const alertCloseButton = within(errorAlert).getByRole("button");
     expect(alertCloseButton).toBeInTheDocument();
 
-    expect(alertCloseButton).toHaveAttribute("aria-label", `Close error alert: ${alert.message}`);
+    expect(alertCloseButton).toHaveAttribute(
```
Comment (ezhangy) [inline frontend/src/components/QuestionInput/ErrorAlert.test.tsx]: [dust] I think the `toHaveAccessibleName` matcher could be used here instead!

File: frontend/src/components/QuestionInput/ErrorAlert.test.tsx [hunk 3]
```diff
@@ -39,22 +40,22 @@ describe("Test the onRemove function", () => {
 
   it("calls the onRemove function with the file upload preview ID when the close button is clicked", async () => {
```
Comment (ezhangy) [inline frontend/src/components/QuestionInput/ErrorAlert.test.tsx]: [dust] should the test name be renamed from 
"calls the onRemove function with the file upload preview ID..."
to 
"calls the onRemove function with the alertType..."?

File: frontend/src/components/QuestionInput/ErrorAlertContainer.tsx [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+      className={`alertContainer display-flex flex-column width-full  ${styles.errorAlertContainer}`}
+    >
+      {Object.entries(alerts).map(([alertType, alertMessage]) => {
+        if (alertMessage != null) {
+          return (
+            <ErrorAlert
+              onRemove={onRemove}
+              message={alertMessage}
+              alertType={alertType}
+              key={alertType}
```
Comment (casewalker) [inline frontend/src/components/QuestionInput/ErrorAlertContainer.tsx:26]: sand - It is not clear to me here whether there will always only be exactly 1 or 0 alerts of the same type. If that is the case, then this is ok, but if there could be 2 or more alerts of the same type, then React could have issues with duplicated keys.

For safety, I would consider changing this to `alertMessage`, or maybe `alertType + alertMessage`.

File: frontend/src/utils/alertUtils.ts [hunk 5]
```diff
@@ -1,7 +1,59 @@
...
+import {
+  MAX_UPLOADED_FILE_SIZE_IN_MB,
+  MAX_UPLOADED_IMAGE_SIZE_IN_MB,
+  truncateFilename,
+} from "./fileUploadUtils";
 
-export interface Alert {
-  message: string;
-  id: string;
+export enum AlertTypes {
```
Comment (ezhangy) [inline frontend/src/utils/alertUtils.ts]: [sand] When creating the types for the AlertBanner, I used an object instead of TS-specific enums: 
```ts
export const AlertType = {
  INFO: "info",
  WARNING: "warning",
  SUCCESS: "success",
  ERROR: "error",
  EMERGENCY: "emergency",
} as const;
```

Thoughts on creating an issue to align on one pattern and refactor?

File: frontend/src/components/QuestionInput/ErrorAlertContainer.test.tsx [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    const mockAlerts: AlertsMap = {
+      [AlertTypes.IMAGE_EXCEEDS_MAX_SIZE]: "alert message 1",
+      [AlertTypes.FILE_EXCEEDS_MAX_SIZE]: "alert message 2",
+      [AlertTypes.PROMPT_NOT_ENTERED]: "alert message 3",
+    };
+
+    const { container } = render(<ErrorAlertContainer onRemove={jest.fn()} alerts={mockAlerts} />);
+
+    Object.entries(mockAlerts).forEach(([alertType, alertMessage]) => {
+      const errorAlert = screen.getByTestId(`${alertType}`);
```
Comment (casewalker) [inline frontend/src/components/QuestionInput/ErrorAlertContainer.test.tsx]: dust - Is this string template necessary? Not sure, but it looks like it could just be `screen.getByTestId(alertType);`

File: frontend/src/components/QuestionInput/ErrorAlertContainer.test.tsx [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+import "@testing-library/jest-dom";
+
+import type { AlertsMap } from "../../utils/alertUtils";
+import { AlertTypes } from "../../utils/alertUtils";
+
+import { ErrorAlertContainer } from "./ErrorAlertContainer";
+expect.extend(toHaveNoViolations);
+
+describe("Test the AlertContainer component", () => {
+  it("correctly renders an alert container containing one error alert", async () => {
```
Comment (casewalker) [inline frontend/src/components/QuestionInput/ErrorAlertContainer.test.tsx:37]: dust - This test looks pretty good, but I am a little concerned that it validates only that the `ErrorAlert` is present, which implies that the `ErrorAlertContainer` is also present, but there is no explicit check for details of the container itself.

Maybe that is ok, but on vibes, it feels like it would be good to validate that the container itself is there too.

File: frontend/src/components/QuestionInput/ErrorAlertContainer.test.tsx [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+
+import "@testing-library/jest-dom";
+
+import type { AlertsMap } from "../../utils/alertUtils";
+import { AlertTypes } from "../../utils/alertUtils";
+
+import { ErrorAlertContainer } from "./ErrorAlertContainer";
+expect.extend(toHaveNoViolations);
+
+describe("Test the AlertContainer component", () => {
```
Comment (casewalker) [inline frontend/src/components/QuestionInput/ErrorAlertContainer.test.tsx]: sand - `AlertContainer` is now `ErrorAlertContainer`. This is where in IVR, we have started doing things like:
```
describe(ErrorAlertContainer.name, () => {
  ...
```
To make sure code changes always sync with their describe blocks.

File: frontend/src/components/QuestionInput/ErrorAlert.tsx [hunk 9]
```diff
@@ -1,68 +1,37 @@
...
-  ERROR_ALERT_TIMEOUT_PERIOD_IN_SECONDS,
-} from "../../utils/alertUtils";
 
 import styles from "./QuestionInput.module.css";
 
 interface ErrorAlertProps {
   onRemove: (id: string) => void;
-  alert: Alert;
+  message: string;
+  alertType: string;
```
Comment (ezhangy) [inline frontend/src/components/QuestionInput/ErrorAlert.tsx:10]: [pebble] I think alertType's type should be constrained by the AlertTypes enum defined in alertUtils.ts. Per the [TS docs](https://www.typescriptlang.org/docs/handbook/enums.html#enums-at-compile-time), I think you could create a string union type from the enum like this: 
```ts
type AlertType = keyof typeof AlertTypes;
```

An additional [dust] here is that there's already an `AlertType` enum/type that exists in the AlertBanner component. Thoughts on having clearer naming for each type to make it easier for us to distinguish between them, maybe `UswdsAlertType` and `ErrorAlertType`?

File: frontend/src/components/QuestionInput/ErrorAlertContainer.test.tsx [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+      expect(errorAlert).toBeInTheDocument();
+
+      const alertText = within(errorAlert).getByText(alertMessage!);
+      expect(alertText).toBeInTheDocument();
+    });
+
+    expect(await axe(container)).toHaveNoViolations();
+  });
+
+  it("correctly renders an alert container containing multiple error alerts", async () => {
```
Comment (casewalker) [inline frontend/src/components/QuestionInput/ErrorAlertContainer.test.tsx:55]: sand - Would there be utility in testing also an empty ErrorAlertContainer?

File: frontend/src/utils/alertUtils.ts (hunk 5 above)
Comment (casewalker) [inline frontend/src/utils/alertUtils.ts]: praise - I love enums ❤️

File: frontend/src/components/QuestionInput/ErrorAlertContainer.test.tsx [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+
+    const { container } = render(<ErrorAlertContainer onRemove={jest.fn()} alerts={mockAlerts} />);
+
+    Object.entries(mockAlerts).forEach(([alertType, alertMessage]) => {
+      const errorAlert = screen.getByTestId(`${alertType}`);
+      expect(errorAlert).toBeInTheDocument();
+
+      const alertText = within(errorAlert).getByText(alertMessage!);
+      expect(alertText).toBeInTheDocument();
+    });
```
Comment (ezhangy) [inline frontend/src/components/QuestionInput/ErrorAlertContainer.test.tsx]: [dust] Very minor comment, but I think the forEach isn't technically needed here since there's only one error alert. Maybe it's a bit cleaner to just expect check on the explicit values: 
```ts
const errorAlert = screen.getByTestId(AlertTypes.IMAGE_EXCEEDS_MAX_SIZE);
...
```

File: frontend/src/components/QuestionInput/ErrorAlertContainer.test.tsx
Comment (ezhangy) [inline frontend/src/components/QuestionInput/ErrorAlertContainer.test.tsx:1]: [dust] For completeness, you could add a test checking that the ErrorAlert isn't rendered when the alert message is `null`.

File: frontend/src/components/QuestionInput/QuestionInput.test.tsx [hunk 12]
```diff
@@ -695,14 +646,47 @@ describe("Test uploading files", () => {
...
+        placeholder={"placeholder"}
+        conversationId={undefined}
+        clearOnSend={false}
+      />
+    );
+
+    await uploadFiles([invalidUploadedFile]);
+
+    await waitFor(() => {
+      expect(screen.getByText(/exceeds .* MB and cannot be uploaded/)).toBeInTheDocument();
```
Comment (ezhangy) [inline frontend/src/components/QuestionInput/QuestionInput.test.tsx]: [dust][praise] cool use of the matcher so the error message isn't specific to the file limit!

File: frontend/src/components/QuestionInput/QuestionInput.test.tsx [hunk 13]
```diff
@@ -695,14 +646,47 @@ describe("Test uploading files", () => {
...
+        disabled={false}
+        placeholder={"placeholder"}
+        conversationId={undefined}
+        clearOnSend={false}
+      />
+    );
+
+    await uploadFiles([invalidUploadedFile]);
+
+    await waitFor(() => {
```
Comment (ezhangy) [inline frontend/src/components/QuestionInput/QuestionInput.test.tsx]: [dust] instead of the `waitFor`, would it also work to just use the async `findBy` instead? 
```ts
 expect(screen.findByText(/exceeds .* MB and cannot be uploaded/)).toBeInTheDocument();
```

File: frontend/src/components/QuestionInput/QuestionInput.test.tsx [hunk 14]
```diff
@@ -927,4 +911,41 @@ describe("Test sending input", () => {
...
+        conversationId={undefined}
+        clearOnSend={false}
+      />
+    );
+
+    await clickSubmitButton();
+
+    await waitFor(() => {
+      expect(
+        screen.getByText(/Please enter a prompt into the text field to continue./)
```
Comment (ezhangy) [inline frontend/src/components/QuestionInput/QuestionInput.test.tsx]: [dust] same comment about using `findBy`

File: frontend/src/utils/alertUtils.ts [hunk 15]
```diff
@@ -1,7 +1,59 @@
...
+export const defaultAlertsMap: AlertsMap = {
+  [AlertTypes.IMAGE_EXCEEDS_MAX_SIZE]: null,
+  [AlertTypes.FILE_EXCEEDS_MAX_SIZE]: null,
+  [AlertTypes.PROMPT_NOT_ENTERED]: null,
+  [AlertTypes.EXCEEDED_FILE_CONTENT_CHARACTER_LIMIT]: null,
+  [AlertTypes.EXCEEDED_PROMPT_CHARACTER_LIMIT]: null,
+  [AlertTypes.FAILED_TO_READ_PDF]: null,
+  [AlertTypes.FAILED_TO_READ_DOCX]: null,
+  [AlertTypes.EXCEEDED_MAX_FILE_COUNT]: null,
+};
```
Comment (casewalker) [inline frontend/src/utils/alertUtils.ts]: sand - There is an argument that it is nice to see each `AlertType` spelled out here, explicitly given a `null` value, however, doing this explicitly is slightly less maintainable, since it will require any new `AlertTypes` entries to be added to both the enum and this map.

I would suggest instead something like:
```
    export const defaultAlertsMap: AlertsMap = Object.fromEntries(
      Object.values(AlertTypes).map((enumValue) => [enumValue, null]),
    );
```

Is that readable enough? I'm not sure, but I kinda like it. (Maybe replace `enumValue` with `alertType`? Up to you!)

File: frontend/jest.setup.tsx (hunk 1 above)
Comment (sannidhishukla) [reply frontend/jest.setup.tsx:12]: yeah this mock + framer motion are still used for testing one other component (not included in this PR) so this is still needed!

File: frontend/src/components/QuestionInput/ErrorAlertContainer.tsx (hunk 4 above)
Comment (sannidhishukla) [reply frontend/src/components/QuestionInput/ErrorAlertContainer.tsx:26]: there should always be exactly 1 or 0 alerts of the same type!

File: frontend/src/components/QuestionInput/ErrorAlertContainer.test.tsx (hunk 6 above)
Comment (sannidhishukla) [reply frontend/src/components/QuestionInput/ErrorAlertContainer.test.tsx]: not needed! updating

File: frontend/src/components/QuestionInput/ErrorAlertContainer.test.tsx (hunk 8 above)
Comment (sannidhishukla) [reply frontend/src/components/QuestionInput/ErrorAlertContainer.test.tsx]: updating the describe block text for now but might start taking the IVR approach in the future

File: frontend/src/components/QuestionInput/ErrorAlertContainer.test.tsx (hunk 10 above)
Comment (sannidhishukla) [reply frontend/src/components/QuestionInput/ErrorAlertContainer.test.tsx:55]: added a test for this!

File: frontend/src/utils/alertUtils.ts (hunk 15 above)
Comment (sannidhishukla) [reply frontend/src/utils/alertUtils.ts]: this is great! i was thinking it felt a little repetitive and a little annoying to maintain if there are more alert types in the future as is...

File: frontend/src/components/QuestionInput/ErrorAlertContainer.tsx (hunk 4 above)
Comment (casewalker) [reply frontend/src/components/QuestionInput/ErrorAlertContainer.tsx:26]: Ah, yes, the type is given as `AlertsMap`, so TS will ensure it is a healthy object without duplicate keys. Sick!!

File: frontend/src/utils/alertUtils.ts [hunk 16]
```diff
@@ -1,7 +1,59 @@
...
+  } else {
+    const filenamesString = filenames
+      .map((fileName) => {
+        return truncateFilename(fileName);
+      })
+      .join(", ");
+
+    return `The following files exceed ${MAX_UPLOADED_FILE_SIZE_IN_MB} MB and cannot be uploaded: ${filenamesString}`;
+  }
+};
```
Comment (casewalker) [inline frontend/src/utils/alertUtils.ts]: pebble - I think for these two helpers, the returned error messages don't have periods at the end. The other error messages which are directly inside QuestionInput do seem to all have periods at the ends. I think these functions should be changed to do the same.

File: frontend/src/components/QuestionInput/QuestionInput.tsx [hunk 17]
```diff
@@ -173,7 +180,6 @@ export const QuestionInput = ({
     if (clearOnSend) {
       setQuestion("");
       setSelectedFiles([]);
-      setInputErrors([]);
```
Comment (casewalker) [inline frontend/src/components/QuestionInput/QuestionInput.tsx:176]: dust - This will now hang on to errors after "send"? Is that the right approach here?

File: frontend/src/components/QuestionInput/QuestionInput.tsx [hunk 18]
```diff
@@ -279,6 +279,7 @@ export const QuestionInput = ({
     const files = event.target.files;
 
     if (files != null) {
+      setInputErrors(defaultAlertsMap);
```
Comment (casewalker) [inline frontend/src/components/QuestionInput/QuestionInput.tsx:290]: dust - So am I right that the input files were changed and now we are going to reset all errors so that we can re-validate the file input and only generate any errors that are true for the current input files?

Is there any concern over removing other types of input errors which may persist, like `EXCEEDED_PROMPT_CHARACTER_LIMIT`?

File: frontend/src/components/QuestionInput/QuestionInput.tsx [hunk 19]
```diff
@@ -327,13 +325,16 @@ export const QuestionInput = ({
     setSelectedFiles((selectedFiles) => selectedFiles.filter((file) => file.fileId !== idToClose));
   };
 
-  const removeError = (idToRemove: string): void => {
-    setInputErrors((inputErrors) => inputErrors.filter((error) => error.id !== idToRemove));
+  const removeError = (alertTypeToRemove: string): void => {
```
Comment (casewalker) [inline frontend/src/components/QuestionInput/QuestionInput.tsx]: sand - It might be more painful, not sure, but I think you could set this type to be `keyof typeof AlertTypes`, thus validating the input a little more tightly?

File: frontend/src/utils/alertUtils.ts (hunk 5 above)
Comment (casewalker) [inline frontend/src/utils/alertUtils.ts]: sand - I think this might be better named as the singular `AlertType`, I think that is what is common practice usually: https://stackoverflow.com/a/15756009/25621843

File: frontend/src/components/QuestionInput/QuestionInput.tsx (hunk 18 above)
Comment (sannidhishukla) [reply frontend/src/components/QuestionInput/QuestionInput.tsx:290]: that is correct! imo there is no concern over removing those errors - if someone tries to submit the prompt that's too long again then the error will reappear on submission

File: frontend/src/components/QuestionInput/QuestionInput.tsx (hunk 17 above)
Comment (sannidhishukla) [reply frontend/src/components/QuestionInput/QuestionInput.tsx:176]: yeah, `inputErrors` is now being set to `defaultAlertsMap` at the top of this function (so the errors are being cleared out regardless of if `clearOnSend` is set.

this would just mean that if `clearOnSend` is set to true and someone unsuccessfully sends the input and gets input errors, those errors aren't immediately cleared out here.

(on that note it might be a good idea to remove `clearOnSend`-related stuff soon. i don't think that's set to true and i don't really see that changing in the future...

File: frontend/src/components/QuestionInput/QuestionInput.test.tsx [hunk 20]
```diff
@@ -388,9 +340,9 @@ describe("Test error alerts", () => {
 
     await uploadFiles([uploadedFileOne, uploadedFileTwo]);
 
-    const inputErrorOne = screen.getByTestId(`errorAlert-exceedsMaxSize-${mockUuidOne}`);
+    const inputErrorOne = screen.getByTestId(`${AlertType.IMAGE_EXCEEDS_MAX_SIZE}`);
```
Comment (casewalker) [inline frontend/src/components/QuestionInput/QuestionInput.test.tsx]: sand - It looks like input1 and 2 and inputAfter1 and 2 can all remove the template steering wrapping.

File: frontend/src/components/QuestionInput/QuestionInput.test.tsx [hunk 21]
```diff
@@ -695,14 +646,47 @@ describe("Test uploading files", () => {
 
     await uploadFiles([invalidUploadedFile, validUploadedFile]);
 
-    const inputError = screen.getByTestId(`errorAlert-exceedsMaxSize-${mockInvalidFileUuid}`);
+    const inputError = screen.getByTestId(`${AlertType.IMAGE_EXCEEDS_MAX_SIZE}`);
```
Comment (casewalker) [inline frontend/src/components/QuestionInput/QuestionInput.test.tsx]: sand - Same comment about template wrapper

File: frontend/src/utils/alertUtils.ts [hunk 22]
```diff
@@ -1,7 +1,52 @@
...
+  if (filenames.length === 1) {
+    return `Image file ${truncateFilename(filenames[0])} exceeds ${MAX_UPLOADED_IMAGE_SIZE_IN_MB} MB and cannot be uploaded.`;
+  } else {
+    const filenamesString = filenames
+      .map((fileName) => {
+        return truncateFilename(fileName);
+      })
+      .join(", ");
+
+    return `The following image files exceed ${MAX_UPLOADED_IMAGE_SIZE_IN_MB} MB and cannot be uploaded: ${filenamesString}`;
```
Comment (casewalker) [inline frontend/src/utils/alertUtils.ts]: sand - I think this one and the similar line in `getFileExceedsMaxSizeErrorMessage` may still benefit from ending with periods.

**Review by casewalker — APPROVED** [review]
I think that's everything! Nice work, with no boulders in the comments, I'm approving!

File: frontend/src/components/QuestionInput/QuestionInput.tsx [hunk 23]
```diff
@@ -102,14 +116,13 @@ export const QuestionInput = ({
...
-      setInputErrors([
-        ...inputErrors,
-        {
-          message: `Please enter a prompt into the text field to continue.`,
-          id: `promptNotEnteredError-${uuidv4()}`,
-        },
-      ]);
+      setInputErrors((prevInputErrors) => ({
+        ...prevInputErrors,
+        [AlertType.PROMPT_NOT_ENTERED]: `Please enter a prompt into the text field to continue.`,
```
Comment (ezhangy) [inline frontend/src/components/QuestionInput/QuestionInput.tsx]: [dust] since there's no string interpolation, we could use single/double quotes instead of backticks here

File: frontend/src/components/QuestionInput/QuestionInput.tsx [hunk 24]
```diff
@@ -239,13 +242,10 @@ export const QuestionInput = ({
...
         }
       } catch (err) {
-        setInputErrors([
-          ...inputErrors,
-          {
-            message: `Could not read text from .docx file: ${truncateFilename(selectedFile.name)}. Please try uploading a different file.`,
-            id: `${selectedFile.name}-failedToReadDocx-${Date.now()}`,
-          },
-        ]);
+        setInputErrors((prevInputErrors) => ({
```
Comment (ezhangy) [inline frontend/src/components/QuestionInput/QuestionInput.tsx]: [dust][for my own understanding] what happens in the case that there are multiple PDFs/docx files in an upload that cannot be read?

File: frontend/src/components/QuestionInput/QuestionInput.tsx [hunk 25]
```diff
@@ -102,14 +116,13 @@ export const QuestionInput = ({
...
+
     if (!question.trim()) {
-      setInputErrors([
-        ...inputErrors,
-        {
-          message: `Please enter a prompt into the text field to continue.`,
-          id: `promptNotEnteredError-${uuidv4()}`,
-        },
-      ]);
+      setInputErrors((prevInputErrors) => ({
```
Comment (ezhangy) [inline frontend/src/components/QuestionInput/QuestionInput.tsx:123]: [dust/sand] if you press send with an empty input multiple times when using a screen reader, the alert is only read the first time since there's no change in the DOM thereafter. (This is also true of all the other alerts, but the PROMPT_NOT_ENTERED seems the most likely to maintain the same message). 

If there's a simple way to flash the alerts off/on each time they're triggered, that would be great! If not, maybe it's something we can do in a follow up ticket?

**Review by ezhangy — APPROVED** [review]
This looks great, some comments but nothing blocking! 

(Note: Testing out how alerts work with a screen reader (Safari + VoiceOver), I'm still noticing trouble with alerts triggered by the file picker being read out loud. I think this is alright though, because since the alerts are no longer timed users will be able to navigate to the error. I think there's a potential enhancement around moving focus to the alert when it's triggered to notify the user, but definitely something we can do in a follow-up ticket!)

File: frontend/src/utils/alertUtils.ts (hunk 5 above)
Comment (sannidhishukla) [reply frontend/src/utils/alertUtils.ts]: that sounds like a good idea for a potential refactor! is there a specific reason that you used `as const` instead of an enum?

File: frontend/src/components/QuestionInput/QuestionInput.tsx (hunk 25 above)
Comment (sannidhishukla) [reply frontend/src/components/QuestionInput/QuestionInput.tsx:123]: i'll add a follow up ticket for this!

File: frontend/src/components/QuestionInput/QuestionInput.tsx (hunk 24 above)
Comment (sannidhishukla) [reply frontend/src/components/QuestionInput/QuestionInput.tsx]: added some changes for the error messaging here!

File: frontend/src/components/QuestionInput/QuestionInput.tsx (hunk 25 above)
Comment (sannidhishukla) [reply frontend/src/components/QuestionInput/QuestionInput.tsx:123]: added [ticket](https://github.com/newjersey/innovation-platform-pm/issues/237)

File: frontend/src/utils/alertUtils.ts (hunk 5 above)
Comment (ezhangy) [reply frontend/src/utils/alertUtils.ts]: I used `as const` at the time because the string enums are a TS-only feature, and I wasn't sure about the full implications of what that means when it's transpiled to JS. I'm not against using them though! `as const` was just the more known quantity for me at the time. 

(BTW, Case wrote a good overview for the options in [this comment](https://github.com/newjersey/nj-ai-assistant/pull/51#discussion_r1996132694) on my last PR that may be helpful!)

File: frontend/src/components/QuestionInput/QuestionInput.test.tsx [hunk 26]
```diff
@@ -553,7 +558,34 @@ describe("Test error alerts", () => {
...
+        conversationId={undefined}
+        clearOnSend={false}
+      />
+    );
+
+    await uploadFiles([uploadedDocxFile, uploadedPdfFile]);
+    await inputChatMessage();
+    await clickSubmitButton();
+
+    expect(screen.getByText(/Could not read text from files:/)).toBeInTheDocument();
```
Comment (ezhangy) [inline frontend/src/components/QuestionInput/QuestionInput.test.tsx:590]: [dust] could we add `expect`s checking for the presence of the filenames in the error message? i don't feel too strongly about this since we're testing the error messages in `alertUtils.test.ts`, but it seems slightly more robust to also verify here that both the filenames appear in the error message.

(this comment applies to the .pdf and .docx specific tests above as well)

File: frontend/src/components/QuestionInput/QuestionInput.test.tsx [hunk 27]
```diff
@@ -70,7 +70,7 @@ describe("Test file upload previews", () => {
     });
   });
 
-  afterEach(jest.clearAllMocks);
+  afterEach(uuidv4Mock.mockReset());
```
Comment (ezhangy) [inline frontend/src/components/QuestionInput/QuestionInput.test.tsx:73]: [dust] (for my own understanding) why the switch from `jest.clearAllMocks` to `uuidv4Mock.mockReset()`?

File: frontend/src/components/QuestionInput/QuestionInput.tsx [hunk 28]
```diff
@@ -131,19 +133,48 @@ export const QuestionInput = ({
...
+          }
+        }
       })
-    );
+    ).then((files) => files.filter((file): file is UploadedFile => file !== null));
+
+    if (filesWithProcessingErrors.length > 0) {
+      setInputErrors((prevInputErrors) => ({
+        ...prevInputErrors,
+        [ErrorAlertType.FAILED_TO_READ_FILE]:
```
Comment (ezhangy) [inline frontend/src/components/QuestionInput/QuestionInput.tsx:154]: [dust/sand] is it worth adding a `logEvent` for the `FAILED_TO_READ_FILE` as well?

**Review by ezhangy — APPROVED** [review]
looks great!! I really like the approach you took for when there are multiple files that can't be read

just a few dusts, nothing blocking!

File: frontend/src/components/QuestionInput/QuestionInput.test.tsx (hunk 27 above)
Comment (sannidhishukla) [reply frontend/src/components/QuestionInput/QuestionInput.test.tsx:73]: basically i noticed that `jest.clearAllMocks` was not actually clearing the mock implementations. it turns out `clearAllMocks` actually just clears information on things like number of calls to the mock, etc. but not the actual implementation (which `mockReset` _**does**_ clear).

here is a [link to some documentation on clearAllMocks](https://jestjs.io/docs/jest-object#jestclearallmocks)

File: frontend/src/components/QuestionInput/QuestionInput.tsx (hunk 28 above)
Comment (sannidhishukla) [reply frontend/src/components/QuestionInput/QuestionInput.tsx:154]: hmm i'm not sure how useful that information would be to log. for other errors that we're adding a `logEvent` for (file size error, input prompt length error, etc.) we could in theory look at the frequency with which we're seeing those errors and make some adjustments on our end based on how often they're showing up (for example, for the `submit_prompt_client_error_file_size` log, we can see if users are, say, really frequently uploading really large PDFs and failing and increase the max PDF upload size we allow if that seems like an important user need).

with this error, it seems more related to the individual files that a user might be uploading that might not have "valid content" which doesn't seem like actionable information for us to have.

is there something that you think would be helpful for us to log from this that i'm missing, though?

Comment (ezhangy) [reply frontend/src/components/QuestionInput/QuestionInput.tsx:154]: That's a good point! To be honest, I was mostly thinking about it in terms of completeness since I noticed we were logging the other errors. I guess if I had to argue for logging the error, I would say that it would help us see if users are running into the error frequently, which might signal we should e.g. add help text around the file uploads.

But I take your point that the data isn't as actionable! Agreed that it doesn't seem like logging would add much value here.
